addons/payment_authorize/models/payment.py

```diff
@@ -161,3 +190,74 @@ def _authorize_form_validate(self, data):
                 'acquirer_reference': data.get('x_trans_id'),
             })
             return False
+
+    @api.multi
+    def authorize_s2s_do_transaction(self, **data):
+        self.ensure_one()
+        transaction = AuthhorizeRequest(self.acquirer_id.environment, self.acquirer_id.authorize_login, self.acquirer_id.authorize_transaction_key)
+        tree = transaction.create_authorize_s2s_transaction(self.payment_token_id.acquirer_ref, self.payment_token_id.authorize_payment_id, self.amount, str(self.reference))
+        return self._authorize_s2s_validate_tree(tree)
+
+    @api.multi
+    def _authorize_s2s_validate_tree(self, tree):
+        return self._authorize_s2s_validate(tree)
+
+    @api.multi
+    def _authorize_s2s_validate(self, tree):
+        self.ensure_one()
+        if self.state == 'done':
+            _logger.warning('Authorize: trying to validate an already validated tx (ref %s)' % self.reference)
+            return True
+        status_code = int(tree.get('x_response_code', '0'))
+        if status_code == self._authorize_valid_tx_status:
+            self.write({
+                'state': 'done',
+                'acquirer_reference': tree.get('x_trans_id'),
```

Comment:
Please set 'date_validate' either to now as well when setting the state to done.